aloha-vw-jni/src/main/scala/com/eharmony/aloha/models/vw/jni/multilabel/VwMultilabelParamAugmentation.scala

```diff
@@ -0,0 +1,453 @@
+    * next matching of a regex, it is consumable by the prefix of a regex padded with
+    * the `pad` function.
+    * @param s a string
+    * @return
+    */
+  private[this] def pad(s: String) = """(^|\s)""" + s + """(?=\s|$)"""
+  private[this] val NumRegex            = """-?(\d+(\.\d*)?|\d*\.\d+)([eE][+-]?\d+)?"""
+  private[this] val ClassCastMsg        = """(\S+) cannot be cast to (\S+)""".r
+  private[this] val CsoaaRank           = pad("--csoaa_rank").r
+  private[this] val WapOrCsoaa          = pad("""--(csoaa|wap)_ldf\s+(mc?)""").r
+  private[this] val Quiet               = pad("--quiet").r
+  protected     val Ignore      : Regex = pad("""--ignore\s+(\S+)""").r
+  protected     val IgnoreLinear: Regex = pad("""--ignore_linear\s+(\S+)""").r
+  private[this] val UnrecoverableFlags  = pad("--(" + UnrecoverableFlagSet.mkString("|") + ")").r
+  private[this] val QuadraticsShort     = pad("""-q\s*([\S]{2})""").r
+  private[this] val QuadraticsLong      = pad("""--quadratic\s+(\S{2})""").r
+  private[this] val Cubics              = pad("""--cubic\s+(\S{3})""").r
+  private[this] val Interactions        = pad("""--interactions\s+(\S{2,})""").r
+  private[this] val NoConstant          = pad("""--noconstant""").r
+  private[this] val ConstantShort       = pad("""-C\s*(""" + NumRegex + ")").r
+  private[this] val ConstantLong        = pad("""--constant\s+(""" + NumRegex + ")").r
+
+
+  private[this] val FlagsToRemove = Seq(
+    QuadraticsShort,
+    QuadraticsLong,
+    Cubics,
+    Interactions,
+    NoConstant,
+    ConstantShort,
+    ConstantLong,
+    CsoaaRank,
+    IgnoreLinear,
+    Ignore
+  )
+
+  /**
+    * Remove flags (and options) for the flags listed in `FlagsToRemove`.
+    * @param vwParams VW params passed to the `updatedVwParams` function.
+    * @return
+    */
+  protected def removeParams(vwParams: String): String =
+    FlagsToRemove.foldLeft(vwParams)((s, r) => r.replaceAllIn(s, ""))
+
+  protected def addParams(
+      paramsAfterRemoved: String,
+      namespaceNames: Set[String],
+      oldIgnored: VWNsSet,
+      oldIgnoredLinear: VWNsSet,
+      oldInteractions: Set[String],
+      labelNs: LabelNamespaces
+  ): String = {
+    val i = oldIgnored + labelNs.dummyLabelNs
+
+    // Don't include namespaces that are ignored in ignore_linear.
+    val il = (toVwNsSet(namespaceNames) ++ oldIgnoredLinear) -- i
+
+    // Don't turn a given namespace into quadratics interacted on label when the
+    // namespace is listed in the ignore_linear flag.
+    val qs = il.flatMap(n =>
+      if (oldIgnored.contains(n) || oldIgnoredLinear.contains(n)) Nil
+      else List(s"${labelNs.labelNs}$n")
+    )
+
+    // Turn quadratic into cubic and cubic into higher-order interactions.
+    val cs  = createLabelInteractions(oldInteractions, oldIgnored, labelNs, _ == 2)
+    val hos = createLabelInteractions(oldInteractions, oldIgnored, labelNs, _ >= 3)
+
+    val quadratics = qs.toSeq.sorted.map(q => s"-q$q" ).mkString(" ")
```

Comment:
You can probably making this simpler by just making `-q`, `--cubic`, and `--interactions` all go to `--interactions` directly.